plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/dialogs/driver/DriverDependenciesTree.java

```diff
@@ -175,6 +180,17 @@ public boolean loadLibDependencies() throws DBException {
         return resolved;
     }
 
+    private void grayOutInstalledArtifact(DBPDriverDependencies.DependencyNode node, TreeItem item) {
+        Path localFile = node.library.getLocalFile();
+        try {
+            if ((localFile != null && Files.exists(localFile) && Files.size(localFile) > 0) && editable) {
```

Comment:
Always perform cheap operations first so it can short-circuit without performing redundant work.

```suggestion
            if (editable && localFile != null && Files.exists(localFile) && Files.size(localFile) > 0) {
```